tools/test.py

```diff
@@ -1713,25 +1655,32 @@ def Main():
         print "Could not create the temporary directory", options.temp_dir
         sys.exit(1)
 
-  if options.report:
-    PrintReport(all_cases)
-
-  result = None
-  def DoSkip(case):
-    # A list of tests that should be skipped can be provided. This is
-    # useful for tests that fail in some environments, e.g., under coverage.
-    if options.skip_tests != [""]:
-        if [ st for st in options.skip_tests if st in case.case.file ]:
-            return True
-    if SKIP in case.outcomes or SLOW in case.outcomes:
+  def should_keep(case):
+    if any([ (s in case.file) for s in options.skip_tests ]):
```

Comment:
Nit: This list is not necessary, `any` should be able to work with any iterable, in this case, a generator expression.

Reply:
ask